app/controllers/api/v8/submissions_controller.rb

```diff
@@ -0,0 +1,283 @@
+        key :description, 'User\'s own submissions in json'
+        schema do
+          key :title, :submissions
+          key :required, [:submissions]
+          property :submissions do
+            key :type, :array
+            items do
+              key :'$ref', :Submission
+            end
+          end
+        end
+      end
+    end
+  end
+
+  swagger_path '/api/v8/courses/{course_id}/exercises/submissions/mine' do
+    operation :get do
+      key :description, 'Returns the user\'s own submissions in a json format'
+      key :operationId, 'findUsersOwnSubmissionsById'
+      key :produces, [
+          'application/json'
+      ]
+      key :tags, [
+          'submission'
+      ]
+      parameter '$ref': '#/parameters/path_course_id'
+      response 403, '$ref': '#/responses/auth_required'
+      response 404 do
+        key :description, 'Course not found'
+        schema do
+          key :title, :errors
+          key :type, :json
+        end
+      end
+      response 200 do
+        key :description, 'User\'s own submissions in json'
+        schema do
+          key :title, :submissions
+          key :required, [:submissions]
+          property :submissions do
+            key :type, :array
+            items do
+              key :'$ref', :Submission
+            end
+          end
+        end
+      end
+    end
+  end
+
+  around_action :course_transaction
+
+  def all_submissions
+    @course = Course.lock('FOR SHARE').find_by(name: "#{params[:slug]}-#{params[:course_name]}") || Course.lock('FOR SHARE').find_by(id: params[:course_id])
+    authorize! :read, @course
+
+    @organization = @course.organization
+    @submissions = Submission.where(course_id: @course.id)
+    filter_submissions(@submissions)
+  end
+
+  def users_submissions
+    @course = Course.lock('FOR SHARE').find_by(name: "#{params[:slug]}-#{params[:course_name]}") || Course.lock('FOR SHARE').find(params[:course_id])
+    authorize! :read, @course
+
+    @user = User.find(params[:user_id])
+    authorize! :read, @user
+
+    @submissions = Submission.where(course_id: @course.id, user_id: @user.id)
+    filter_submissions(@submissions)
+  end
+
+  def my_submissions
+    @user = current_user
+    authorize! :read, @user
+
+    @course = Course.lock('FOR SHARE').find_by(name: "#{params[:slug]}-#{params[:course_name]}") || Course.lock('FOR SHARE').find(params[:course_id])
+    authorize! :read, @course
+
+    @submissions = Submission.where(course_id: @course.id, user_id: @user.id)
+    filter_submissions(@submissions)
+  end
+
+  private
+
+  def course_transaction
+    Course.transaction(requires_new: true) do
+      yield
+    end
+  end
+
+  def render_json(array)
+    if array.empty?
+      render json: {
+        error: "You are not signed in!"
+      }
+    else
+      render json: {
+        submissions: array
+      }
+    end
+  end
+
+  def filter_submissions(subs)
+    visible_submissions = []
+    subs.each do |submission|
```

Comment:
As this looks like something common to do, I would this move this to the Submission model. It could be something along the lines of:
```ruby
def self.readable
  select { |submission| submission.readable_by?(current_user) }
end
```